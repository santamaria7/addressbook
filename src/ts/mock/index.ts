import configureMockStore from "redux-mock-store";
import createSagaMiddleware from "redux-saga";
import rootSaga from "../store/saga";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
export function mockStore(){
    configure({ adapter: new Adapter() });
    const sagaMiddleware = createSagaMiddleware();
    const mockStoreAPI = configureMockStore([sagaMiddleware]);

    const store = mockStoreAPI({
        users: [],
        pagination: {
            limit: 1000,
            offset: 1,
        },
        search: {
            isFiltered: false,
        },
        loading: false,
        nationality: "",
    });
    sagaMiddleware.run(rootSaga);
    return store
}