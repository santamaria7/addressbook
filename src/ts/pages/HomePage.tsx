import React from "react";
import UsersList from "../components/UsersList";

const HomePage = () => {
  return <UsersList />
};

export default HomePage;
