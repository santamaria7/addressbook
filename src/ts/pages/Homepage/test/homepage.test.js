import renderer from "react-test-renderer";
import HomePage from "../index";
import React from "react";
import { Provider } from "react-redux";
import { mockStore } from "../../../../../mock";
import { shallow } from "enzyme";

describe("HomePage", function () {
  it("should render home page - with snapshot", function () {
    const component = renderer.create(
      <Provider store={mockStore()}>
        <HomePage />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render home page - without snapshot", function () {
    const wrapper = shallow(
      <Provider store={mockStore()}>
        <HomePage />
      </Provider>
    );
    expect(wrapper.find("HomePage").length).toEqual(1);
  });
});
