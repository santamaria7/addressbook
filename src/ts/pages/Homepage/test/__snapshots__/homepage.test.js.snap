// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render home page - with snapshot 1`] = `
<div
  className="page"
>
  <h1
    className="page__title"
  >
    My Address Book
  </h1>
  <div
    className="search "
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="firstName"
        >
          First name: 
        </label>
        <input
          id="firstName"
          name="first"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="lastName"
        >
          Last name: 
        </label>
        <input
          id="lastName"
          name="last"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Go
      </button>
    </form>
  </div>
  <div
    className="users"
  />
</div>
`;
